src/data-gateway/search-data-gateway.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  EVENTS_FILE_ENV,
+  NEWS_FILE_ENV,
+  PROGRAM_FILE_ENV,
+  PROJECT_FILE_ENV,
+} from 'config';
+import { type SearchDetailInput } from 'entities';
+import { type IFileService } from 'services/file-services/flat-file-types';
+import { type ISearchDataGateway } from 'usecases';
+
+const fileName = [
+  NEWS_FILE_ENV,
+  EVENTS_FILE_ENV,
+  PROGRAM_FILE_ENV,
+  PROJECT_FILE_ENV,
+];
+
+export class SearchDataGateway implements ISearchDataGateway {
+  constructor(private readonly fileService: IFileService) {}
+
+  async fetch(data: SearchDetailInput): Promise<any[]> {
+    const fileNameCategory = this.getFileNameForCategory(data.category);
+
+    try {
+      var results: any[] = [];
```

Comment:
The logic in this should actually be in the search usecase. 

Here is the idea. The search usecase should have all these other gateways as its dependencies(ie the constructor).